pkg/reconciler/apibinding/apibinding_reconcile.go

```diff
@@ -290,16 +298,27 @@ func (r *workspaceAPIExportReferenceReconciler) reconcile(ctx context.Context, a
 		// The crd was deleted and needs to be recreated. `existingCRD` might be non-nil if
 		// the lister is behind, so explicitly set to nil to ensure recreation.
 		if r.deletedCRDTracker.Has(crd.Name) {
-			klog.Infof("Bound CRD %s|%s was deleted - need to recreate", ShadowWorkspaceName, crd.Name)
+			klog.V(4).Infof("Bound CRD %s|%s was deleted - need to recreate", ShadowWorkspaceName, crd.Name)
 			existingCRD = nil
 		}
 
 		if existingCRD == nil {
 			// Create flow
+
+			// Check for naming conflicts
+			if err := r.nameConflictChecker.checkForConflicts(crd, logicalcluster.From(apiBinding)); err != nil {
+				conditions.MarkFalse(
+					apiBinding,
+					apisv1alpha1.CRDReady,
+					apisv1alpha1.NamingConflictsReason,
+					conditionsv1alpha1.ConditionSeverityError,
+					err.Error(),
+				)
+				return reconcileStatusStop, nil // don't retry
```

Comment:
(No action required) I'm not sure what to make of the persistent `// don't retry` comments. I think it may be as important to not return an error to ensure the status change (`conditions.MarkFalse`) is persisted?

Reply:
Good point about an error precluding the status patching. I have adjusted the reconciler to always patch (open for discussion)